### ISSUE TITLE
Weird behavior of null character

### ISSUE THREAD

**Issue by stoarca** [mscdex/node-pcre#3]
Please see the following test cases for some weird behavior involving null characters. I haven't looked through your code yet, but I suspect it has something to do with how the strings are passed to the pcre library (null-terminated).

```
PCRE.exec('\\0', '\0', 0); // returns null, should return [0, 1]
PCRE.exec('a\\0a', 'a\0a', 0); // returns null, should return [0, 3]
```

**Comment by mscdex** [comment]
This is a bug with upstream PCRE as they do not provide an API that supports passing in the pattern length (just the c-string).

**Comment by mscdex** [comment]
Alternatively what you can do is use something like `'a\\x00a'` for your pattern.

I am in the process of fixing the problem with nulls in the subject ( **UPDATE:** now fixed in 40eb15d).

**Comment by stoarca** [comment]
Thanks for the quick fix!
